### PR TITLE
rpm: fix missing %preun and upgrade regression

### PR DESCRIPTION
**Pull request by kenhys**
%preun section and upgrade sequence wasn't migrated from omnibus-td-agent.

### CODE REVIEW

**Comment by kenhys** [comment]
Oops, I'll check upgrade sequence again.

```
$ sudo rpm -Uvh td-agent-3.7.1-1.el8.x86_64.rpm 
Verifying...                          ################################# [100%]
Preparing...                          ################################# [100%]
Updating / installing...
   1:td-agent-3.7.1-1.el8             ################################# [ 50%]
/var/tmp/rpm-tmp.IfN8Ai: line 1: 2: command not found
Cleaning up / removing...
   2:td-agent-3.7.0-0.el8             ################################# [100%]
```

**Comment by kenhys** [comment]
It seems that /sbin path issue.

File: td-agent/yum/td-agent.spec.in [hunk 1]
```diff
@@ -25,6 +25,14 @@
 %define __arch_install_post %{nil}
 %undefine __brp_mangle_shebangs
 
+%if %{use_systemd}
+%define control_service() (systemctl %1 %2)
```
Comment (cosmo0920) [inline td-agent/yum/td-agent.spec.in:29]: Should we use full path here?

Comment (kenhys) [reply td-agent/yum/td-agent.spec.in:29]: No, systemctl is placed at /usr/bin/, so no need to use full path here.

**Comment by kenhys** [comment]
Upgrade is ok on CentOS 6:

```
$ rpm -q td-agent
td-agent-3.7.0-0.el6.x86_64
[vagrant@localhost ~]$ sudo rpm -Uvh td-agent-3.7.1-1.el6.x86_64.rpm
Preparing...                ########################################### [100%]
   1:td-agent               ########################################### [100%]
   $ sudo yum removge td-agent
   読み込んだプラグイン:fastestmirror
   コマンド「removge」が見つかりません。「/usr/bin/yum --help」を実行してください。
   [vagrant@localhost ~]$ sudo yum remove td-agent
   読み込んだプラグイン:fastestmirror
   削除処理の設定をしています
   依存性の解決をしています
   --> トランザクションの確認を実行しています。
   ---> Package td-agent.x86_64 0:3.7.1-1.el6 will be 削除
   --> 依存性解決を終了しました。

   依存性を解決しました

   =============================================================================================================================================================
    パッケージ                           アーキテクチャ                     バージョン                              リポジトリー                           容量
   =============================================================================================================================================================
   削除:
    td-agent                             x86_64                             3.7.1-1.el6                             installed                              47 M

   トランザクションの要約
   =============================================================================================================================================================
   削除                 1 パッケージ

   インストール済み容量: 47 M
   これでいいですか? [y/N]y
   パッケージをダウンロードしています:
   rpm_check_debug を実行しています
   トランザクションのテストを実行しています
   トランザクションのテストを成功しました
   トランザクションを実行しています
   警告: RPMDB は yum 以外で変更されました。
   Stopping td-agent: td-agent[  OK  ]
     削除中                  : td-agent-3.7.1-1.el6.x86_64                                                                                                  1/1
     Verifying               : td-agent-3.7.1-1.el6.x86_64                                                                                                  1/1

   削除しました:
     td-agent.x86_64 0:3.7.1-1.el6                                                                                                                              

   完了しました!
```

**Comment by kenhys** [comment]
Fresh install is ok on CentOS 6

```
   $ sudo rpm -ivh td-agent-3.7.1-1.el6.x86_64.rpm
   Preparing...                ########################################### [100%]
      1:td-agent               ########################################### [100%]
   [vagrant@localhost ~]$ sudo yum remove td-agent-3.7.1-1.el6.x86_64.rpm
   読み込んだプラグイン:fastestmirror
   削除処理の設定をしています
   引数に一致しません: td-agent-3.7.1-1.el6.x86_64.rpm
   Loading mirror speeds from cached hostfile
    * base: ftp-srv2.kddilabs.jp
    * extras: ftp-srv2.kddilabs.jp
    * updates: ftp-srv2.kddilabs.jp
   削除と設定したパッケージはありません。
   [vagrant@localhost ~]$ sudo yum remove td-agent
   読み込んだプラグイン:fastestmirror
   削除処理の設定をしています
   依存性の解決をしています
   --> トランザクションの確認を実行しています。
   ---> Package td-agent.x86_64 0:3.7.1-1.el6 will be 削除
   --> 依存性解決を終了しました。

   依存性を解決しました

   =============================================================================================================================================================
    パッケージ                           アーキテクチャ                     バージョン                              リポジトリー                           容量
   =============================================================================================================================================================
   削除:
    td-agent                             x86_64                             3.7.1-1.el6                             installed                              47 M

   トランザクションの要約
   =============================================================================================================================================================
   削除                 1 パッケージ

   インストール済み容量: 47 M
   これでいいですか? [y/N]y
   パッケージをダウンロードしています:
   rpm_check_debug を実行しています
   トランザクションのテストを実行しています
   トランザクションのテストを成功しました
   トランザクションを実行しています
   警告: RPMDB は yum 以外で変更されました。
   Stopping td-agent: td-agent[  OK  ]
     削除中                  : td-agent-3.7.1-1.el6.x86_64                                                                                                  1/1
     Verifying               : td-agent-3.7.1-1.el6.x86_64                                                                                                  1/1

   削除しました:
     td-agent.x86_64 0:3.7.1-1.el6                                                                                                                              

   完了しました!
```

**Comment by kenhys** [comment]
Upgrade is ok on CentOS 7(systemd)

```
$ sudo rpm -Uvh td-agent-3.7.1-1.el7.x86_64.rpm
Preparing...                          ################################# [100%]
Updating / installing...
   1:td-agent-3.7.1-1.el7             ################################# [ 50%]
Cleaning up / removing...
   2:td-agent-3.7.0-0.el7             ################################# [100%]
```

**Comment by kenhys** [comment]
Remove is ok on CentOS 7(systemd)

```
   $ sudo rpm -ivh td-agent-3.7.1-1.el7.x86_64.rpm
   Preparing...                          ################################# [100%]
   Updating / installing...
      1:td-agent-3.7.1-1.el7             ################################# [100%]
   [vagrant@localhost ~]$ sudo yum remove td-agent
   Failed to set locale, defaulting to C
   Loaded plugins: fastestmirror
   Resolving Dependencies
   --> Running transaction check
   ---> Package td-agent.x86_64 0:3.7.1-1.el7 will be erased
   --> Finished Dependency Resolution

   Dependencies Resolved

   =============================================================================================================================================================
    Package                              Arch                               Version                                 Repository                             Size
   =============================================================================================================================================================
   Removing:
    td-agent                             x86_64                             3.7.1-1.el7                             installed                              48 M

   Transaction Summary
   =============================================================================================================================================================
   Remove  1 Package

   Installed size: 48 M
   Is this ok [y/N]: y
   Downloading packages:
   Running transaction check
   Running transaction test
   Transaction test succeeded
   Running transaction
   Warning: RPMDB altered outside of yum.
     Erasing    : td-agent-3.7.1-1.el7.x86_64                                                                                                               1/1
     Verifying  : td-agent-3.7.1-1.el7.x86_64                                                                                                               1/1

   Removed:
     td-agent.x86_64 0:3.7.1-1.el7                                                                                                                              

   Complete!
```

**Pull request merged**